forest/src/daemon.rs

```diff
@@ -344,20 +347,27 @@ pub(super) async fn start(config: Config) {
 
     // Block until ctrl-c is hit
     ctrlc_oneshot.await.unwrap();
+    info!("Line {}, refs: {}", line!(), db_weak_ref.strong_count());
 
     let keystore_write = task::spawn(async move {
         keystore.read().await.flush().unwrap();
     });
+    info!("Line {}, refs: {}", line!(), db_weak_ref.strong_count());
 
     // Cancel all async services
-    prometheus_server_task.cancel().await;
-    head_changes_task.cancel().await;
-    republish_task.cancel().await;
+    prometheus_server_task.abort();
```

Comment:
maybe let's put a lambda `cancel_task if it's feasible? Seems like some code duplication that we can get rid of

Reply:
Done.